test/tbb/test_concurrent_queue.cpp

```diff
@@ -281,3 +282,63 @@ TEST_CASE("Test clear and dtor with TrackableItem") {
     test_tracking_dtors_on_clear<oneapi::tbb::concurrent_queue<TrackableItem>>();
     test_tracking_dtors_on_clear<oneapi::tbb::concurrent_bounded_queue<TrackableItem>>();
 }
+
+template<typename CQ, typename T>
+void test_queue_helper(){
+    int size = 5;
+    T vec_1(size, 0), vec_2(size, 0), vec_3(size, 0), vec_4(size, 0);
+    srand(static_cast<unsigned>(time(0)));
+    generate(vec_1.begin(), vec_1.end(), rand);
+    generate(vec_2.begin(), vec_2.end(), rand);
+    generate(vec_3.begin(), vec_3.end(), rand);
+    generate(vec_4.begin(), vec_4.end(), rand);
+    
+    CQ q1, q2;
+    CQ q4({vec_1, vec_2, vec_3});
+    CQ q3 = {vec_4, vec_2, vec_3};
+
+    q1 = q3;
+    q2 = std::move(q3);
+
+    CHECK(q1 != q4);
+    q1.swap(q4);
+    CHECK(q2 == q4);
+    CHECK(q2 == q3);
```

Comment:
After line 301, `q3` should be empty and not equal to `q2`. Maybe `CHECK(q3.empty())`?

Reply:
If allocator is equal, I just modify the pointer. So q3 is not empty in the end, but q2 == q3 is also irrelevant - removed it.